lib/shopify_api/rest/resources/2023_01/gift_card_adjustment.rb

```diff
@@ -16,7 +16,7 @@ class GiftCardAdjustment < ShopifyAPI::Rest::Base
     def initialize(session: ShopifyAPI::Context.active_session)
       super(session: session)
 
-      @amount = T.let(nil, T.nilable(Float))
+      @amount = T.let(nil, T.nilable(String))
```

Comment:
I believe float is the right type, we just need to cast it as such if we are provided a string as the argument.

Reply:
@nelsonwittwer Thanks for taking a look at this!

Agree a numeric type would make sense - would `BigDecimal` be preferable to `Float` though for a current amount?